pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -325,6 +328,14 @@ def add_qiskit_data(self, data: "QuantumCircuitData") -> None:
                 q_ctrl_box = QControlBox(c_box, instr.num_ctrl_qubits)
                 self.tkc.add_qcontrolbox(q_ctrl_box, qubits)
 
+            elif isinstance(instr, Initialize):
+                # Check that the Initialize object is constructed with a list:
+                # https://qiskit.org/documentation/stubs/qiskit.circuit.QuantumCircuit.initialize.html
+                if isinstance(instr.params, list):
```

Comment:
There is a minor complication here if `Initialize` is not constructed using a python list or numpy array.

https://qiskit.org/documentation/stubs/qiskit.circuit.QuantumCircuit.initialize.html

If Initialize is constructed using a string or an integer instead it can't easily be handled by `StatePreperationBox`. I thought it best to check for this case explictly

Maybe we should add an error message for when the `instr.params` is not a list of amplitudes